browser/ui/views/infobars/brave_wayback_machine_infobar_contents_view.h

```diff
@@ -58,7 +62,7 @@ class BraveWaybackMachineInfoBarContentsView
   void HideInfobar();
 
   void FetchURLButtonPressed();
-  void DontAskButtonPressed();
+  void OnCheckboxUpdated();
```

Comment:
Q. When we use checkbox, do we still want to hide the infobar as soon as user clicks the checkbox? The issue says `We should use a checkbox for consistency and minimize mis-clicks ` then, I feel like we shouldn't hide infobar and give users a chance to unset the checkbox.

Reply:
This PR works likes you commented - https://github.com/brave/brave-core/pull/19876/files#diff-d4adcb39c22a83f7fc71ee6c47663498a672fcd1c6eee7d59fe3bfa1b3291e88L129

Reply:
Oops, my bad. Thanks for the link